pkgs/unified_analytics/lib/src/session.dart

```diff
@@ -70,6 +63,16 @@ class Session {
     return _sessionId;
   }
 
+  /// Preps the [Session] class with the data found in the session json file.
+  ///
+  /// We must check if telemetry is enabled to refresh the session data
+  /// because the refresh method will write to the session file and for
+  /// users that have opted out, we have to leave the session file empty
+  /// per the privacy document
+  void initialize(bool telemetryEnabled) {
```

Comment:
it worries me anytime there is an `initialize()` function that one must remember to call at the right time, because a refactor could easily lead to code now accessing state in this class BEFORE the `initialize()` method was called.

I think this would solve the particular issue linked, and maybe this is the right change to make immediately to stop crashes, but I think long-term we should refactor all of these classes (especially `Analytics`) such that it is not possible to access any of its state BEFORE it has been initialized.